### ISSUE TITLE
RCS1212 false positive

### ISSUE THREAD

**Issue by sm-g** [JosefPihrt/Roslynator#584]
**Product and Version Used**: 
roslynator 2.1.3 for VS2019

**Steps to Reproduce**:
```csharp
TransformBlock<TimeSpan, T> block = null;
block = new TransformBlock<TimeSpan, T>(async delayTime =>
{
	await Task.Delay(delayTime, cancellationToken)
		.ConfigureAwait(false);

	block.Post(delayTime);

	return action();
}, new ExecutionDataflowBlockOptions
{
	CancellationToken = cancellationToken
});

return block;
```

**Actual Behavior**:
RCS1212 on second line (`return new TransformBlock...`)

**Expected Behavior**:
Nothing

**Comment by JosefPihrt** [comment]
Thanks for the report @sm-g !
